contracts/Multi-signature/ApprovalCollector.sol

```diff
@@ -0,0 +1,215 @@
+pragma solidity ^0.4.24;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "./MultiplexingContract.sol";
+
+
+/**
+ * @title DelegateERC20 smart contract
+ * @dev Controls execution of ERC20 token transactions
+ */
+contract ApprovalCollector is Ownable {
+    using SafeMath for uint256;
+
+    MultiplexingContract private multiplexingContract;             //address of controlled contract
+    uint256 constant CALL_LIFETIME = 300000;    //transaction lifetime in seconds
+    uint256 private _transactionNonce = 0;
+
+    enum ContractClass {DEFAULT, CRITICAL}
+
+    struct ContractClassParams {
+        uint neededApprovals;
+        uint neededCriticalApprovals;
+    }
+
+    struct Transaction {
+        mapping (address => bool) approvers;
+        uint256 timestamp;
+        uint arrayIndex;    //index in array of pending transactions
+
+        ContractClassParams classParams;
+
+        bytes transaction;  //function selector + encoded arguments
+        address executor;   //contract, that should perform call
+    }
+
+    bytes32[] _pendingTransactions;
+    mapping (bytes32 => Transaction) private _transactions;
+
+    mapping (address => bool) private _administrators;
+    mapping (address => bool) private _criticalApprovers;
```

Comment:
I think the code would be more readable if the names had more symmetry. For example, `administrators - criticalAdministrators` or `approvers - criticalApprovers`.